sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -387,6 +390,101 @@ case class CatalogStatistics(
   }
 }
 
+/**
+ * This class of statistics for a column is used in [[CatalogTable]] to interact with metastore.
+ */
+case class CatalogColumnStat(
+  distinctCount: Option[BigInt] = None,
+  min: Option[String] = None,
+  max: Option[String] = None,
+  nullCount: Option[BigInt] = None,
+  avgLen: Option[Long] = None,
+  maxLen: Option[Long] = None,
+  histogram: Option[Histogram] = None) {
+
+  /**
+   * Returns a map from string to string that can be used to serialize the column stats.
+   * The key is the name of the column and name of the field (e.g. "colName.distinctCount"),
+   * and the value is the string representation for the value.
+   * min/max values are stored as Strings. They can be deserialized using
+   * [[ColumnStat.fromExternalString]].
```

Comment:
nit: Shall we move `fromExternalString` from `ColumnStat` to `CatalogColumnStat`?

Reply:
I think that actually everything from ColumnStat object should move.
`fromExternalString` / `toExternalString` -> `CatalogColumnStat`

And also:
`supportsDatatype` / `supportsHistogram` -> `AnalyzeColumnCommand`
`statExprs` / `rowToColumnStat` -> `AnalyzeColumnCommand`
because they are tied to that specific method of stats collection.